vali_objects/position.py

```diff
@@ -518,13 +527,23 @@ def _clamp_and_validate_leverage(self, order) -> bool:
 
         is_first_order = len(self.orders) == 0
         proposed_leverage = self.net_leverage + order.leverage
+        proposed_portfolio_leverage = net_portfolio_leverage + (order.leverage * self.trade_pair.leverage_multiplier)
         min_position_leverage, max_position_leverage = leverage_utils.get_position_leverage_bounds(self.trade_pair, order.processed_ms)
-        if abs(proposed_leverage) > max_position_leverage:
-            if is_first_order or abs(proposed_leverage) >= abs(self.net_leverage):
-                order.leverage = max(0.0, max_position_leverage - abs(self.net_leverage))
-                if self.position_type == OrderType.SHORT:
+        max_portfolio_leverage = leverage_utils.get_portfolio_leverage_cap(order.processed_ms)
+
+        if (abs(proposed_leverage) > max_position_leverage
```

Comment:
we need to give the community ample time to prepare for this change. Sam please make an announcement in the Discord. I can proof read beforehand if you'd like.